lib/util/src/bin_heap.c

```diff
@@ -472,7 +472,7 @@ bin_heap2_insert_src(struct bin_heap2 *bh, struct element_source *es)
 {
     struct heap_node *node;
     void *            elt;
-    int               i, width;
+    int               width;
```

Comment:
There are a lot of interesting integer shenanigans going on in the bin heap code. For instance, we take max_width as a uint32_t but store it as an int on the bin heap struct.

Might tackle it after the spill bin heap PR